evennia/utils/dbserialize.py

```diff
@@ -644,7 +644,15 @@ def process_item(item):
                 # classmethods) and instances
                 pass
 
-        if hasattr(item, "__iter__"):
+        if hasattr(item, "items"):
```

Comment:
I'm  a little reluctant to base item identification on the existence of an `items` property; it's not unthinkable that a user may add a property `.items` to their non-dict entity (to represent a nested inventory, for example). Using dunder methods and duck typing seems better. I don't think there's a unique dunder for dict (`__getitem__` is in both list and dict), but maybe something like this? (pseudocode):

```
if item has __iter__ property: 
    try to process key, val as if a dictionary 
    if failing to do so (iterating does not generate key,val pairs):
        try it as a list
```

Reply:
looks good to me!

Reply:
done